File: python/day11.py
from pprint import pprint
from collections import defaultdict


def count_stones(stones):
    return sum([c for _,c in stones.items()])

def day11(input):

    stones = defaultdict(lambda:0)
    for s in [int(s) for s in input.split()]:
        stones[s] += 1

    print(stones)

    for i in range(75):
        newstones = defaultdict(lambda:0)
        for s,c in stones.items():
            if s == 0:
                newstones[1] += c
            else:
                x = str(s)
                if len(x) % 2 == 0:
                    newstones[int(x[:len(x)//2])] += c
                    newstones[int(x[len(x)//2:])] += c
                else:
                    newstones[s * 2024] += c
        stones = newstones
        if i == 24:
            acc1 = count_stones(stones)
    acc2 = count_stones(stones)
    return acc1,acc2

input = open("src/bin/inputs/day11.test1.txt", "r").read()
pprint(day11(input.strip()))


input = open("src/bin/inputs/day11.txt", "r").read()
pprint(day11(input.strip()))

File: src/bin/day11.rs
const INPUT: &str = include_str!("inputs/day11.txt");

pub fn day11(input: &str) -> (usize, usize) {
    (input.len(), input.len())
}

fn main() {
    let (p1, p2) = day11(INPUT);
    println!("{}\n{}", p1, p2);
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test1() {
        let input = include_str!("inputs/day11.test1.txt");
        let (p1, p2) = day11(input);

        assert_eq!(p1, input.len());
        assert_eq!(p2, input.len());
    }

    #[test]
    fn test_main() {
        let (p1, p2) = day11(INPUT);

        assert_eq!(p1, INPUT.len());
        assert_eq!(p2, INPUT.len());
    }
}
